File: packages/sdk/typescript/human-protocol-sdk/example/worker.ts
/* eslint-disable no-console */
import { ChainId, OrderDirection } from '../src/enums';
import { WorkerUtils } from '../src/worker';

export const getWorkers = async () => {
  const workers = await WorkerUtils.getWorkers({
    chainId: ChainId.POLYGON_AMOY,
    first: 100,
    skip: 0,
    orderBy: 'payoutCount',
    orderDirection: OrderDirection.DESC,
  });

  console.log('Filtered workers:', workers);

  const worker = await WorkerUtils.getWorker(
    ChainId.POLYGON_AMOY,
    workers[0].address
  );

  console.log('Worker details:', worker);
};

(async () => {
  await getWorkers();
})();

File: packages/sdk/python/human-protocol-sdk/human_protocol_sdk/gql/worker.py
from human_protocol_sdk.filter import WorkerFilter

worker_fragment = """
fragment WorkerFields on Worker {
    id
    address
    totalHMTAmountReceived
    payoutCount
}
"""


def get_worker_query() -> str:
    return """
query GetWorker($address: String!) {{
    worker(id: $address) {{
        ...WorkerFields
    }}
}}
{worker_fragment}
""".format(
        worker_fragment=worker_fragment
    )


def get_workers_query(filter: WorkerFilter) -> str:
    return """
query GetWorkers(
    $address: String
    $orderBy: String
    $orderDirection: String
    $first: Int
    $skip: Int
) {{
    workers(
        where: {{
            {address_clause}
        }}
        orderBy: $orderBy
        orderDirection: $orderDirection
        first: $first
        skip: $skip
    ) {{
        ...WorkerFields
    }}
}}
{worker_fragment}
""".format(
        worker_fragment=worker_fragment,
        address_clause=("address: $address" if filter.worker_address else ""),
    )
